### ISSUE TITLE
mapbox-gl-leaflet no longer works with v1.14.0

### ISSUE THREAD

**Issue by codespree** [maplibre/maplibre-gl-js#123]
## Motivation
[mapbox-gl-leaflet](https://github.com/mapbox/mapbox-gl-leaflet) no longer works with release [v1.14.0](https://github.com/maplibre/maplibre-gl-js/releases/tag/v1.14.0) of maplibre-gl-js

This is due to the fact that:
```
new mapboxgl.Map(options);
```
changed to:
```
new maplibregl.Map(options);
```

### Implementation

I have created a fork:
https://github.com/codespree/maplibre-gl-leaflet

Here are three working examples:
[Basic example](https://raw.githack.com/codespree/maplibre-gl-leaflet/aa06e210ab8f8ea136407a8f1481b4b8077c3d5a/examples/basic.html)
[Cluster example](https://raw.githack.com/codespree/maplibre-gl-leaflet/aa06e210ab8f8ea136407a8f1481b4b8077c3d5a/examples/cluster.html)
[Map events example](https://raw.githack.com/codespree/maplibre-gl-leaflet/aa06e210ab8f8ea136407a8f1481b4b8077c3d5a/examples/events.html)

I would like to transfer the repository to maplibre for easier maintenance.

**Comment by wipfli** [comment]
#79

**Comment by nyurik** [comment]
I'm ok to transfer it here, but we would probably want to push it as a new git so that there is no "forked from" link back to the original. @codespree once your project is ready, i will simply load/push it to maplibre, and will add you to the maintainers of it.

**Comment by codespree** [comment]
@nyurik Actually, since its basically a single `js` file:
https://github.com/codespree/maplibre-gl-leaflet/blob/master/leaflet-maplibre-gl.js

Maybe you want to just include it in this repository directly.

You can also include the examples:
https://github.com/codespree/maplibre-gl-leaflet/tree/master/examples

**Comment by gknapp** [comment]
Will this be published as an npm module too, `maplibre-gl-leaflet`?

**Comment by jgravois** [comment]
given the speed at which you've introduced unnecessary breaking changes and started forking dependent libraries, its probably time to revist the **Roadmap** and **Avoid Fragmentation** sections of your README.

> Roadmap
> The primary goal is consistency and _backwards-compatability with previous releases_...

> Avoid Fragmentation
> ... we all benefit from "one project" rather than "our way". If you know of other forks, please reach out to them and direct them here.

Personally i don't care if you'd rather fork `mapbox-gl-leaflet` instead of opening a small PR to make it cross-compatible, but i think it would be helpful to your users if you were honest about what your priorities are.

---

i'm sure ☝️ comes off as pretty harsh, but tbh i'm having a hard time understanding why you can't promote this project's new name and simultaneously do the work outlined in https://github.com/maplibre/maplibre-gl-js/issues/33 to make migrating to your own fork less painful for folks.

just my two cents, carry on as you will. 🤗

**Comment by wipfli** [comment]
Thanks for your feedback @jgravois. I can understand your frustration and I have already encountered the problems which you describe with other community members as well. 

I think it is a good idea to update the README and give step by step instructions of what needs to be changed if one wants to go from a `mapbox v1.13.0` to a `maplibre v1.14.0` project.

**Comment by wipfli** [comment]
> Maybe you want to just include it in this repository directly.

I think it is cleaner to have a separate repository for this, for example https://github.com/maplibre/maplibre-gl-leaflet.

**Comment by wipfli** [comment]
> Will this be published as an npm module too, maplibre-gl-leaflet?

Yes, absolutely :) It will appear in the `@maplibre` organization on NPM.

**Comment by wipfli** [comment]
@codespree thanks for preparing `maplibre-gl-leaflet` on your GitHub account. I have created a repo for this in the `maplibre` organization now:

https://github.com/maplibre/maplibre-gl-leaflet

Please go ahead and open a pull request on this repository with your code.

**Comment by wipfli** [comment]
Were you thinking of something like this https://github.com/maplibre/maplibre-gl-js/pull/148 @jgravois ?

**Comment by jgravois** [comment]
#148 is definitely better than what you had before, but y'all could sidestep forking every mapbox-gl-js plugin ever created if this library just continued to create a bundle that exported to the old namespace.

**Comment by animaux** [comment]
> but y'all could sidestep forking every mapbox-gl-js plugin ever created if this library just continued to create a bundle that exported to the old namespace.

MapLibre-Folks, you really should think twice about this. @jgravois makes a very valid point here.

**Comment by wipfli** [comment]
Thanks @animaux for joining the discussion. 

Do I understand you correctly that your suggestion would be to use `mapbox-gl.js` as the file name for the bundle and `mapbox` as the javascript object name? 

Maybe you can elaborate on how you think MapLibre should look like.

**Comment by animaux** [comment]
I’m not a programmer, but as @jgravois pointed out, the change of namespaces to `maplibre` pretty much breaks compatibility with ALL mapbox-gl plugins. That’s a huge step away. If it’s possible to have namespace-compatibility for both `maplibre` and `mapbox`, that should be the way to go, I feel.

**Comment by jgravois** [comment]
i'm not suggesting an additional plugin. i'm suggesting something that was already outlined in #33, using rollup to generate and ship an _additional_ UMD bundle with the original namespace. (see https://rollupjs.org/guide/en/#outputname).

if you did that users and plugin authors could opt into the new namespace at their leisure.

regardless of whether you do that or not, opening PRs for existing plugins to help them fallback on the `maplibregl` namespace when `mapboxgl` is undefined would relieve you of the burden of maintaining a growing number of forks for the sake of your name change.

**Comment by HarelM** [comment]
While I think this discussion is super important and interesting I suggest two things:
1. Someone can try and write an example to demonstrate how this would work, I have no clue how to do this, honestly. (Considering AoT and tree shaking, this becomes even harder I think)
2. Move this discussion to the discussions area as I think this particular issue is resolved.
I'm in favor of closing this issue and continue this discussion in the discussions...
